libs/designsystem/src/lib/helpers/deep-merge.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Performs a deep merge of objects and returns new object. Does not modify
+ * objects (immutable).
+ * It merges from into object to the left
+ *  eg. mergeDeep({a: 'will be overridden', b: 'some val'}, {a: 'new val'}) =>  {a: 'new val', b: 'some val'}
+ *
+ * @param toOverride object - Object to override,
+ * @param overrides object - Object to override "toOverride" values with
+ * @param config object - Configuration for how to merge.
+ *  mergeArrays determines if it should merge or override arrays
+ * @returns {object} New object with merged key/values
+ */
+export function mergeDeep(toOverride, overrides, config = { mergeArrays: false }) {
+  const isObject = (obj) => obj && typeof obj === 'object';
+
+  return [toOverride, overrides].reduce((combinedObject, obj) => {
+    Object.keys(obj).forEach((key) => {
+      const combinedObjectValue = combinedObject[key];
+      const currentObjectValue = obj[key];
+
+      if (Array.isArray(combinedObjectValue) && Array.isArray(currentObjectValue)) {
+        combinedObject[key] = config.mergeArrays
+          ? combinedObjectValue.concat(currentObjectValue)
+          : [...currentObjectValue];
```

Comment:
In our case, we don't want to concat values as eg. the data array will grow otherwise when merge is called multiple times.

Reply:
?

Reply:
In our case, we want an array to replace a corresponding array in the combined object. Eg. we don't want a merge of series.data values.

Reply:
Removed the spread